src/Relude/Extra/Foldable1.hs

```diff
@@ -88,9 +88,11 @@ instance Foldable1 NonEmpty where
     fold1 = sconcat
     {-# INLINE fold1 #-}
 
-    foldMap1 :: Semigroup m => (a -> m) -> NonEmpty a -> m
-    foldMap1 f (a :| [])     = f a
-    foldMap1 f (a :| b : bs) = f a <> foldMap1 f (b :| bs)
+    foldMap1 :: forall m a . Semigroup m => (a -> m) -> NonEmpty a -> m
+    foldMap1 f (a :| as) = foldr go f as a
```

Comment:
@josephcsible Could you, please, add one more `doctest` example of non-commutative operation to the `foldMap1` definition inside `Foldable1` typeclass? Just to make sure that the new implementation indeed works as expected.

Reply:
Done.